### PR TITLE
Add godoc to generated params

### LINKED ISSUE #15: doc comments are not generated in type definitions
The types generated in api-params.go have no doc comments; it would be great if all the
exported identifiers could have appropriate doc comments taken from the description
fields in the YAML.

### PR DESCRIPTION
**Pull request by mhemmings**
Closes #15

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -240,11 +249,13 @@ func schemaRefParse(oasSchema *openapi3.SchemaRef, name string) templates.Defini
 		}
 		def := schemaRefParse(r.SchemaRef, name)
 		def.TypeStr = r.Name
+		def.DocComment = "" // We want to avoid duplication of comments due to references.
```
Comment (rogpeppe) [inline main.go]: s/We want to avoid/Avoid/
?

File: templates/template.go [hunk 2]
```diff
@@ -79,3 +80,16 @@ Outer:
 	err := ioutil.WriteFile(filepath, outputData, 0666)
 	return errgo.Mask(err)
 }
+
+// Comment takes a string and turns it into a golang comment using "//", while preserving new lines.
+func Comment(str string) string {
```
Comment (rogpeppe) [inline templates/template.go:84]: or:

	func Comment(str string) string {
		str = strings.TrimSpace(str)
		if str == "" {
			return ""
		}
		return "// " + strings.Replace(str, "\n", "\n// ", -1)
	}
?

File: testdata/component-naming.txt [hunk 3]
```diff
@@ -39,4 +39,6 @@ type Error struct {
 	Code ErrorCode `json:"code"`
 }
 
+// A code representing the kind of error.
+// TODO enumerate all the possible error codes here?
```
Comment (rogpeppe) [inline testdata/component-naming.txt]: Probably better to leave this TODO out - it's perhaps a bit misleading.

File: main.go [hunk 4]
```diff
@@ -114,15 +114,22 @@ func main2() error {
 				// Ignore (https://github.com/go-httprequest/httprequest/blob/2b21a94c9e788981d4e609ef4b7a21cedae6da66/type.go#L225)
 				continue
 			}
+			name := strcase.ToCamel(op.OperationID + "Request")
 			req := templates.Definition{
-				Name: strcase.ToCamel(op.OperationID + "Request"),
+				Name:       name,
+				DocComment: templates.Comment(fmt.Sprintf("%s holds the request to a %s call.", name, op.OperationID)),
```
Comment (rogpeppe) [inline main.go]: %s holds the request value for a %s call.

?

**Review by rogpeppe — APPROVED** [review]
This is great, thanks!
LGTM modulo a few comments and suggestions.

File: main.go (hunk 4 above)
Comment (mhemmings) [reply main.go]: What do you mean by this?

Comment (rogpeppe) [reply main.go]: I'm suggesting an alternative wording for the comment.

"FooRequest holds the request to a Foo call" doesn't quite read right to me.

**Pull request merged**